uktrade/data-hub-frontend
# Refactor to make redis client and reporter (raven) reusable

## Description of change

There should only need to be one redis client that gets re-used (or at least the code to create the client)
This removes the `redis-store` and adds a `redis-client` which is used to get a client
The `redis-client` is then used by `session-store` to create the redis backed store
The `redis-client` should be responsible for reporting errors so the raven setup needed to be reusable, so raven was remove from server.js and put in a new module called `reporter`

## Test instructions

This is really just a refactor to help with the next phase so nothing should change. Ensure errors are still reported to Sentry (or logged to the console if a DSN is not present) and that all code that uses Redis still works.

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied. 
https://github.com/uktrade/data-hub-frontend/blob/develop/docs/Code%20review%20guidelines.md"
